metakb/query.py

```diff
@@ -307,27 +331,336 @@ def search(self, variation='', disease='', therapy='', gene='',
+                    else:
+                        if sb_id.startswith('civic.eid'):
+                            supported_by.append(sb_id)
+                except ValueError:
+                    sb_not_found.add(sb_id)
+            if sb_not_found:
+                response['warnings'].append(f"Supported by evidence not "
+                                            f"yet  supported in MetaKB: "
+                                            f"{sb_not_found} for "
+                                            f"{s['id']}")
+
+            params = {
+                'id': s_id,
+                'description': s.get('description'),
+                'direction': s.get('direction'),
+                'evidence_level': s.get('evidence_level'),
+                'variation_origin': s.get('variation_origin'),
+                'proposition': proposition,
+                'variation_descriptor': variation_descr,
+                'therapy_descriptor': therapy_descr,
+                'disease_descriptor': disease_descr,
+                'method': method,
+                'supported_by': supported_by
+            }
+            response['statements'].append(
+                NestedStatementResponse(**params).dict())
+            added_statements.add(s_id)
+        session.close()
+        return SearchStatementsService(**response).dict(
+            by_alias=True, exclude_none=True)
+
+    def _add_to_proposition_cache(self, session, p_node,
+                                  proposition_cache: Dict) -> None:
+        """Add a proposition to `proposition_cache`
+
+        :param session: Session
+        :param p_node: Proposition Node
+        :param Dict proposition_cache: Proposition lookup dictionary
+        """
+        p_id = p_node.get('id')
+        if p_id not in proposition_cache:
+            proposition_resp = session.read_transaction(
+                self._find_and_return_proposition_response,
+                p_id
+            )
+            proposition_type = p_node.get('type')
+            proposition = {
+                'id': p_id,
+                'type': proposition_type,
+                'predicate': p_node.get('predicate'),
+                'subject': proposition_resp['subject'],
+                'object_qualifier': proposition_resp['object_qualifier']
+            }
+            if proposition_type == PropositionType.PREDICTIVE:
+                proposition['object'] = proposition_resp['object']
+                proposition = \
+                    TherapeuticResponseProposition(**proposition)
+            elif proposition_type == PropositionType.PROGNOSTIC:
+                proposition = PrognosticProposition(**proposition)
+            elif proposition_type == PropositionType.DIAGNOSTIC:
+                proposition = DiagnosticProposition(**proposition)
+            else:
+                # TODO Check
+                raise Exception(f"{proposition_type} is not a valid "
+                                f"proposition type")
```

Comment:
maybe this should be a ValueError or something more specific?

Reply:
Lol left the todo comment to change this but forgot about it